openmdao/core/total_jac.py

```diff
@@ -1519,6 +1540,12 @@ def check_total_jac(self, raise_error=True, tol=1e-16):
         if np.any(col):  # there's at least 1 row that's zero across all columns
             zero_rows = []
             for name, tup in self.of_meta.items():
+
+                # Support for constraint aliases.
+                if name in self.responses and 'path' in self.responses[name] and \
```

Comment:
This same logic appears in a bunch of places around the code.  Might be worth replacing with a function.